src/pybind/mgr/cephadm/serve.py

```diff
@@ -1292,19 +1289,15 @@ def _get_container_image_info(self, image_name: str) -> ContainerInspectInfo:
         return r
 
     # function responsible for logging single host into custom registry
-    def _registry_login(self, host: str, url: Optional[str], username: Optional[str], password: Optional[str]) -> Optional[str]:
-        self.log.debug(f"Attempting to log host {host} into custom registry @ {url}")
+    def _registry_login(self, host: str, registry_json: str) -> Optional[str]:
```

Comment:
```suggestion
    def _registry_login(self, host: str, registry_json: Dict[str,str]) -> Optional[str]:
```
Let's avoid serializing and deserializing things internally.